src/structurizr/view/view_set.py

```diff
@@ -91,19 +91,71 @@ def __init__(
     ) -> None:
         """Initialize a view set."""
         super().__init__(**kwargs)
-        # self.enterprise_context_views = set(enterprise_context_views)
-        self.system_landscape_views: Set[SystemLandscapeView] = set(
-            system_landscape_views
+        all_views = chain(
+            system_landscape_views,
+            system_context_views,
+            container_views,
+            component_views,
+            deployment_views,
+            dynamic_views,
+            filtered_views,
         )
-        self.system_context_views: Set[SystemContextView] = set(system_context_views)
-        self.container_views: Set[ContainerView] = set(container_views)
-        self.component_views: Set[ComponentView] = set(component_views)
-        self.deployment_views: Set[DeploymentView] = set(deployment_views)
-        self.dynamic_views: Set[DynamicView] = set(dynamic_views)
-        self.filtered_views: Set[FilteredView] = set(filtered_views)
+        self._views = {view.key: view for view in all_views}
         self.configuration = Configuration() if configuration is None else configuration
         self.set_model(model)
 
+    @property
+    def system_landscape_views(self) -> Iterable[SystemLandscapeView]:
+        """Return the SystemLandscapeViews in this ViewSet."""
+        return (
+            view
+            for view in self._views.values()
+            if isinstance(view, SystemLandscapeView)
+        )
```

Comment:
You could refactor all of these by introducing a helper method:

```py
    def _get_typed_views(self, klass: T) -> Iterable[T]:
        return (
            view
            for view in self._views.values()
            if isinstance(view, klass)
        )
```

(see https://docs.python.org/3/library/typing.html#generics) and then

```suggestion
        return self._get_typed_views(SystemLandscapeView)
```

etc.

Reply:
OK, so I've committed something like this, but it wouldn't compile for me just using T so I've used "T" - is this the right answer or is there a better way?